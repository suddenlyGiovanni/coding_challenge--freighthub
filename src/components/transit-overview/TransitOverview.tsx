import * as React from 'react'
import styled from '@emotion/styled/macro'

import { Cargo } from 'features/shipments/typings'
import { Typography } from '@material-ui/core'
import ArrowForwardIcon from '@material-ui/icons/ArrowForwardIos'

const OverviewContainer = styled.div`
  display: flex;
  align-items: center;
  flex-flow: row wrap;
  flex-direction: row;
  justify-content: space-between;

  padding: 15px;

  border: solid thin lightgray;
  border-radius: 4px;

  @media (max-width: 600px) {
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
  }
`

const ElementContainer = styled.div`
  flex: 1;

  max-width: 170px;
`

const CopyContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  max-width: 170px;
  min-height: 50px;
  padding: 4px;

  border: solid thin lightgray;
  border-radius: 4px;
  background-color: lightgray;
`
const Copy = styled(Typography)`
  text-align: center;
`

const CargoList = styled.ul`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-evenly;

  list-style-type: unset;

  margin-block-start: unset;
  margin-block-end: unset;
  margin-inline-start: unset;
  margin-inline-end: unset;
  padding-inline-start: unset;
`

const CargoElement = styled.li`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;

  width: 100%;
  margin: 5px 0;
  padding: 5px;

  text-align: center;

  border: solid thin grey;
  border-radius: 4px;
`

export interface Props {
  origin: string
  destination: string
  cargo: Cargo[]
}
export const TransitOverview: React.FC<Props> = ({
  origin,
  destination,
  cargo,
}) => {
  return (
    <OverviewContainer>
      <ElementContainer>
        <CopyContainer>
          <Copy>{origin}</Copy>
        </CopyContainer>
      </ElementContainer>

      <ArrowForwardIcon />

      <ElementContainer>
        <CargoList>
          {cargo.map(({ description, type, volume }, idx) => (
            <CargoElement key={idx}>
              <Typography variant="body2">
                {`${description} - ${volume}vol`}
              </Typography>
            </CargoElement>
          ))}
        </CargoList>
      </ElementContainer>

      <ArrowForwardIcon />

      <ElementContainer>
        <CopyContainer>
          <Copy>{destination}</Copy>
        </CopyContainer>
      </ElementContainer>
    </OverviewContainer>
  )
}
