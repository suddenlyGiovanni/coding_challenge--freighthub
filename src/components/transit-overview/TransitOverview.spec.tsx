import React from 'react'
import { render } from '@testing-library/react'

import { TransitOverview, Props } from './TransitOverview'

describe('TransportMode', () => {
  const props: Props = {
    origin: 'Shanghai Port',
    destination: 'SaarbrÃ¼cker Str. 38, 10405 Berlin',
    cargo: [
      {
        type: 'Fabric',
        description: '1000 Blue T-shirts',
        volume: 2,
      },
      {
        type: 'Fabric',
        description: '2000 Green T-shirts',
        volume: 3,
      },
    ],
  }

  test('snapshot', () => {
    const { container } = render(<TransitOverview {...props} />)
    expect(container).toMatchSnapshot()
  })
})
