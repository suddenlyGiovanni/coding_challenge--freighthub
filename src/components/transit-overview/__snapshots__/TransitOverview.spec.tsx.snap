// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransportMode snapshot 1`] = `
<div>
  <div
    class="css-1gzjc7l-OverviewContainer eu79uap0"
  >
    <div
      class="css-1nd1wvf-ElementContainer eu79uap1"
    >
      <div
        class="css-b182ra-CopyContainer eu79uap2"
      >
        <p
          class="MuiTypography-root css-1polke6-Copy eu79uap3 MuiTypography-body1"
        >
          Shanghai Port
        </p>
      </div>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    <div
      class="css-1nd1wvf-ElementContainer eu79uap1"
    >
      <ul
        class="css-xy20c4-CargoList eu79uap4"
      >
        <li
          class="css-bru13b-CargoElement eu79uap5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            1000 Blue T-shirts - 2vol
          </p>
        </li>
        <li
          class="css-bru13b-CargoElement eu79uap5"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            2000 Green T-shirts - 3vol
          </p>
        </li>
      </ul>
    </div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
    <div
      class="css-1nd1wvf-ElementContainer eu79uap1"
    >
      <div
        class="css-b182ra-CopyContainer eu79uap2"
      >
        <p
          class="MuiTypography-root css-1polke6-Copy eu79uap3 MuiTypography-body1"
        >
          Saarbr√ºcker Str. 38, 10405 Berlin
        </p>
      </div>
    </div>
  </div>
</div>
`;
